### PR TITLE
MacOS and tvOS support

### PR DESCRIPTION
**Pull request by ashare80**

### CODE REVIEW

**Review by benpious — COMMENTED** [review]
Thanks for doing this! Only blocker I can see so far is the leftover `print` statement, otherwise it looks great.

I might do a quick test on MacOS before I stamp and merge.

**Comment by benpious** [comment]
I tested this, and found two issues:

1. the mac version doesn't support setting the drawable after it's been laid out. The fix seems to be `needsLayout = true` in `drawableLayers` or `drawable`
2. It seems that MacOS is far less liberal than iOS about calling `getRed:blue:green:alpha:` on colors which use different color spaces, and will throw an exception if this is called on a color that isn't in sRGB. 

I'm okay landing with these still in, I can clean them up later. Just a heads up in case you're using this code.

File: Cyborg/SwiftUIView.swift [hunk 1]
```diff
@@ -45,16 +45,42 @@ import SwiftUI
...
     
     /// Initializer.
     public init(_ drawable: VectorDrawable) {
         self.drawable = drawable
     }
+}
+
+#if os(macOS)
+extension VectorDrawableView: NSViewRepresentable {
+    public typealias NSViewType = VectorView
```
[the remark this answers is not in the document]
Comment (ashare80) [reply Cyborg/SwiftUIView.swift:61]: I don't think I can because the method name is different `makeNSView` vs `makeUIView`

**Comment by ashare80** [comment]
> I tested this, and found two issues:
> 
> 1. the mac version doesn't support setting the drawable after it's been laid out. The fix seems to be `needsLayout = true` in `drawableLayers` or `drawable`
> 2. It seems that MacOS is far less liberal than iOS about calling `getRed:blue:green:alpha:` on colors which use different color spaces, and will throw an exception if this is called on a color that isn't in sRGB.
> 
> I'm okay landing with these still in, I can clean them up later. Just a heads up in case you're using this code.

Cool yeah I didn't really test yet, main goal was just trying to get it to at least import nicely with a macOS/iOS SwiftUI app. I guess technically with the NSView being used the min macOS version could be lower, just not sure how low.

**Pull request merged**